### ISSUE TITLE
Referencing external 3rd party DLL

### ISSUE THREAD

**Issue by 3goats** [tjanczuk/edge#62]
Hi,

I having problems referencing a external DLL from edge. I've added the DLL to the .NET GAL but I still cannot reference from within edge. 

``` javascript
var edge = require('edge');

var hello = edge.func(function () {/*

 var listCertificates = edge.func(function() {/*
 #r "System.dll"
 #r "BouncyCastle.Crypto.dll"

 using Org.BouncyCastle.Pkcs;
 using Org.BouncyCastle.Asn1.X509;
 using Org.BouncyCastle.Crypto.Generators;
 using Org.BouncyCastle.Crypto;
 using Org.BouncyCastle.Security;
 using Org.BouncyCastle.OpenSsl;
 using System.Text;
 using System.IO;

 async (data) =>
 {

 //Requested Certificate Name
 X509Name name = new X509Name("CN=Client Cert, C=NL");

 */});
```

This is the error! 

``` console
Uncaught System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: Unable to compile C# code.
----> Errors when compiling as a CLR library:
error CS0006: Metadata file 'BouncyCastle.Crypto.dll' could not be found
----> Errors when compiling as a CLR async lambda expression:
error CS0006: Metadata file 'BouncyCastle.Crypto.dll' could not be found
   at EdgeCompiler.CompileFunc(IDictionary`2 parameters)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at ClrFunc.Initialize(Handle<v8::Value>* , Arguments* args) in c:\edge-master-2\edge-master\src\clrfunc.cpp:line 91 

```

**Comment by shivasurya** [comment]
Im too getting same error pls someone help 
```
var edge = require('edge');

var helloWorld = edge.func({assemblyFile:'MR6100Api.dll',typeName:'MR6100Api.MR6100Api',
	methodName: 'isNetWorkConnect',
	references: [
        ("C:\\Users\\killer\\Documents\\hello\\sample\\MR6100Api.dll")
    ] 
});

helloWorld('127.0.0.1', function (error, result) {
    if (error) throw error;
    console.log(result);
});
```
Exception
```
    return edge.initializeClrFunc(options);
                ^
Error: Could not load file or assembly 'file:///C:\Users\killer\Documents\hello\sample\MR6100Api
.dll' or one of its dependencies. An attempt was made to load a program with an incorrect format.
    at Error (native)
    at Object.exports.func (C:\Users\killer\Documents\hello\sample\node_modules\edge\lib\edge.js
:154:17)
    at Object.<anonymous> (C:\Users\killer\Documents\hello\sample\server.js:3:23)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
    at startup (node.js:129:16)
    at node.js:814:3
```

please help me out

**Comment by tsiddikee** [comment]
I have done strong name snk for the dll and then added the dll to GAC. It worked perfect.
I have also added the dll to nodejs exe directory. 
So far did follow your instruction ebove. It did not work.
Would be nice if you would create a documentation more clearer than fragmented information.
I saw your first page documentation in github, which does not cover external dll integration.
It would be a big help if you would explain step by step how to work with external dll and c# code from associated dll.
